### PR TITLE
use log level when draining LoggerMailbox, #21895

### PR DESCRIPTION
**Pull request by patriknw**
Refs #21895

### CODE REVIEW

File: akka-actor/src/main/scala/akka/event/LoggerMailbox.scala [hunk 1]
```diff
@@ -33,13 +35,21 @@ private[akka] class LoggerMailbox(owner: ActorRef, system: ActorSystem)
...
+      val logLevel = system.eventStream.logLevel
       var envelope = dequeue
       // Drain all remaining messages to the StandardOutLogger.
       // cleanUp is called after switching out the mailbox, which is why
       // this kind of look works without a limit.
       while (envelope ne null) {
-        // Logging.StandardOutLogger is a MinimalActorRef, i.e. not a "real" actor
-        Logging.StandardOutLogger.tell(envelope.message, envelope.sender)
+        // skip logging if level is OFF
+        if (Logging.AllLogLevels.contains(logLevel))
```
Comment (drewhk) [inline akka-actor/src/main/scala/akka/event/LoggerMailbox.scala]: you can cache this query at the start of the method

**Review by drewhk — APPROVED** [review]
LGTM

**Comment by ktoso** [comment]
Applied Endre's feedback, will merge after validation. 
LGTM

**Review by ktoso — APPROVED** [review]

**Comment by patriknw** [comment]
thanks a lot!

**Pull request merged**